src/lib/importer/hoyoLabFormatParser.tsx

```diff
@@ -0,0 +1,334 @@
+import { Constants, Parts, Sets, SubStats } from '../constants.ts'
+import { v4 as uuidv4 } from 'uuid'
+import { Utils } from '../utils'
+import { RelicAugmenter } from '../relicAugmenter'
+import { RelicEnhance, RelicGrade } from 'types/Relic.js'
+
+export function hoyolabParser(input) {
+  const json = JSON.parse(input)
+  const output: {
+    metadata: {
+      trailblazer: string
+      current_trailblazer_path: string
+    }
+    characters: {
+      characterEidolon: number
+      characterId: string
+      characterLevel: number
+      lightCone: string | null
+      lightConeLevel: number
+      lightConeSuperimposition: number
+    }[]
+    relics: {
+      enhance: RelicEnhance
+      equippedBy: string
+      grade: RelicGrade
+      id: string
+      part: string
+      set: string
+      main: {
+        stat: string
+        value: number
+      }
+      substats: {
+        stat: SubStats
+        value: number
+        addedRolls: number
+      }[]
+      verified: boolean
+    }[]
+  } = {
+    metadata: {
+      trailblazer: 'Stelle',
+      current_trailblazer_path: 'Destruction',
+    },
+    characters: [],
+    relics: [],
+  }
+  for (let i = 0; i < json.data.avatar_list.length; i++) {
+    const character = json.data.avatar_list[i]
+    if (character.equip != null) {
+      output.characters.push({
+        characterEidolon: character.rank,
+        characterId: (character.id).toString(),
+        characterLevel: character.level,
+        lightCone: (character.equip.id).toString(),
+        lightConeLevel: character.equip.level,
+        lightConeSuperimposition: character.equip.rank,
```

Comment:
Does this handle characters without light cones? I assume character.equip would be null